capsules/src/rf233.rs

```diff
@@ -799,48 +891,85 @@ impl<'a, S: spi::SpiMasterDevice + 'a> radio::Radio for RF233<'a, S> {
     fn set_transmit_client(&self, client: &'static radio::TxClient) {
         self.tx_client.set(Some(client));
     }
+
     fn set_receive_client(&self, client: &'static radio::RxClient, buffer: &'static mut [u8]) {
         self.rx_client.set(Some(client));
         self.rx_buf.replace(buffer);
     }
 
+    fn set_config_client(&self, client: &'static radio::ConfigClient) {
+        self.cfg_client.set(Some(client));
+    }
+
     fn set_receive_buffer(&self, buffer: &'static mut [u8]) {
         self.rx_buf.replace(buffer);
     }
 
+    fn config_set_address(&self, addr: u16) {
+        self.addr.set(addr);
+    }
+
+    fn config_set_pan(&self, id: u16) {
+        self.pan.set(id);
+    }
 
-    // Setting the address also sets the panx
-    fn set_address(&self, addr: u16) -> ReturnCode {
-        let state = self.state.get();
-        // The start state will push addr into hardware on initialization;
-        // the ready state needs to do so immediately.
-        if state == InternalState::READY || state == InternalState::START {
-            self.addr.set(addr);
-            if state == InternalState::READY {
-                self.state_transition_write(RF233Register::SHORT_ADDR_0,
-                                            (self.addr.get() & 0xff) as u8,
-                                            InternalState::CONFIG_SHORT0_SET);
-            }
+    fn config_set_tx_power(&self, power: i8) -> ReturnCode {
+        if (power > 4 || power < -17) {
+            ReturnCode::EINVAL
+        } else {
+            self.tx_power.set(power);
+            ReturnCode::SUCCESS
+        }
+    }
+
+    fn config_set_channel(&self, chan: u8) -> ReturnCode {
+        if chan >= 11 && chan <= 26 {
+            self.channel.set(chan);
             ReturnCode::SUCCESS
         } else {
-            ReturnCode::EBUSY
+            ReturnCode::EINVAL
         }
     }
-    // Setting the PAN also sets the address
-    fn set_pan(&self, addr: u16) -> ReturnCode {
-        let state = self.state.get();
-        // The start state will push addr into hardware on initialization;
-        // the ready state needs to do so immediately.
-        if state == InternalState::READY || state == InternalState::START {
-            self.pan.set(addr);
-            if state == InternalState::READY {
+
+
+    fn config_address(&self) -> u16 {
+        self.addr.get()
+    }
+    /// The 16-bit PAN ID
+    fn config_pan(&self) -> u16 {
+        self.pan.get()
+    }
+    /// The transmit power, in dBm
+    fn config_tx_power(&self) -> i8 {
+        self.tx_power.get()
+    }
+    /// The 802.15.4 channel
+    fn config_channel(&self) -> u8 {
+        self.channel.get()
+    }
+
+    fn config_commit(&self) -> ReturnCode {
```

Comment:
Because then you have to store that state in the caller (if you don't want to change them) or do a bunch of get() operations to fill in the current values.